stdlib/printexc.ml

```diff
@@ -43,32 +43,39 @@ let fields x =
   | 2 -> sprintf "(%s)" (field x 1)
   | _ -> sprintf "(%s%s)" (field x 1) (other_fields x 2)
 
-let to_string x =
+let use_printers x =
   let rec conv = function
     | hd :: tl ->
         (match try hd x with _ -> None with
-        | Some s -> s
+        | Some s -> Some s
         | None -> conv tl)
```

Comment:
Stylistically, this could more neatly be:

```ocaml
let result = (try hd x with _ -> None) in
if result <> None then result else conv tl
```
or
```ocaml
  (try
     hd x
   with _ -> conv tl)
```

Reply:
(I'm not asking for a change to the PR because I think the current version is okay but) personally I found the earlier version more readable: it's easier for me to read a pattern-matching on an option type and detect the success and failure case than to understand what `if result <> None` is doing.

A third option would be to use the `match .. with exception` feature to avoid the intermediate `try .. with`.

```ocaml
match hd x with
| None | exception _ -> conv tl
| Some s -> Some s
```

if one is disturbed by the lack of sharing in the last case (I'm not), it's also possible to write it as

```ocaml
| (Some _) as result -> result
```

Reply:
Since I had to fix documentation, I went ahead and went with the third option, I agree it is more readable